relays/ethereum/src/substrate_sync_loop.rs

```diff
@@ -0,0 +1,196 @@
+// Copyright 2019-2020 Parity Technologies (UK) Ltd.
+// This file is part of Parity Bridges Common.
+
+// Parity Bridges Common is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity Bridges Common is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity Bridges Common.  If not, see <http://www.gnu.org/licenses/>.
+
+use crate::ethereum_client::{self, EthereumConnectionParams, EthereumSigningParams};
+use crate::ethereum_types::Address;
+use crate::substrate_client::{self, SubstrateConnectionParams};
+use crate::substrate_types::{
+	Hash, Header, Number, QueuedSubstrateHeader, SubstrateHeaderId, SubstrateHeadersSyncPipeline,
+};
+use crate::sync::{HeadersSyncParams, TargetTransactionMode};
+use crate::sync_loop::{SourceClient, TargetClient};
+use crate::sync_types::SourceHeader;
+use futures::future::{ready, FutureExt, Ready};
+use std::{future::Future, pin::Pin};
+
+/// Interval (in ms) at which we check new Substrate headers when we are synced/almost synced.
+const SUBSTRATE_TICK_INTERVAL_MS: u64 = 10_000;
+/// Interval (in ms) at which we check new Ethereum blocks.
+const ETHEREUM_TICK_INTERVAL_MS: u64 = 5_000;
+
+/// Substrate synchronization parameters.
+#[derive(Debug)]
+pub struct SubstrateSyncParams {
+	/// Ethereum connection params.
+	pub eth: EthereumConnectionParams,
+	/// Ethereum signing params.
+	pub eth_sign: EthereumSigningParams,
+	/// Ethereum bridge contract address.
+	pub eth_contract_address: Address,
+	/// Substrate connection params.
+	pub sub: SubstrateConnectionParams,
+	/// Synchronization parameters.
+	pub sync_params: HeadersSyncParams,
+}
+
+impl Default for SubstrateSyncParams {
+	fn default() -> Self {
+		SubstrateSyncParams {
+			eth: Default::default(),
+			eth_sign: Default::default(),
+			// the address 0x731a10897d267e19b34503ad902d0a29173ba4b1 is the address
+			// of the contract that is deployed by default signer and 0 nonce
+			eth_contract_address: "731a10897d267e19b34503ad902d0a29173ba4b1"
+				.parse()
+				.expect("address is hardcoded, thus valid; qed"),
+			sub: Default::default(),
+			sync_params: HeadersSyncParams {
+				max_future_headers_to_download: 8,
```

Comment:
Please move hardcoded values to constants. I think these kind of parameters could be even grouped in a separate module to have a nice overview of hardcoded configuration.

Reply:
See: https://github.com/paritytech/parity-bridges-common/issues/75